pkg/jobrunaggregator/jobrunaggregatorlib/util.go

```diff
@@ -112,15 +132,110 @@ func WaitAndGetAllFinishedJobRuns(ctx context.Context,
 			case <-time.After(10 * time.Minute):
 				continue
 			case <-ctx.Done():
-				return finishedJobRuns, unfinishedJobRuns, finishedJobRunNames, unfinishedJobRunNames, ctx.Err()
+				return ctx.Err()
 			}
 		}
 
 		break
 	}
+	return nil
+}
+
+type ClusterJobRunWaiter struct {
+	ProwJobClient      *prowjobclientset.Clientset
+	TimeToStopWaiting  time.Time
+	ProwJobMatcherFunc ProwJobMatcherFunc
+	nextLogTime        time.Time
+}
+
+func (w *ClusterJobRunWaiter) allProwJobFinished(allItems []*prowv1.ProwJob) bool {
+	uncompletedJobMap := map[string]*prowv1.ProwJob{}
+	for _, prowJob := range allItems {
+		if !w.ProwJobMatcherFunc(prowJob) {
+			continue
+		}
+		if prowJob.Status.CompletionTime != nil {
+			continue
+		}
+
+		jobRunID := prowJob.Labels[prowJobJobRunIDLabel]
+		uncompletedJobMap[jobRunID] = prowJob
+	}
+	if len(uncompletedJobMap) == 0 {
+		return true
+	}
+	logrus.Infof("Current time %+v and next lgo time: [%v]", time.Now(), w.nextLogTime)
```

Comment:
Logrus includes the current time with your message. 

Coming into this loop every few minutes, I don't see why we wouldn't log every time we came through, it won't hurt anything and it helps us always see what the code is doing. I would simplify and ditch this concept.